lib/ansible/executor/task_executor.py

```diff
@@ -1010,16 +1010,19 @@ def _set_connection_options(self, variables, templar):
 
         task_keys = self._task.dump_attrs()
 
-        # set options with 'templated vars' specific to this plugin and dependant ones
+        # set options with 'templated vars' specific to this plugin and dependent ones
         self._connection.set_options(task_keys=task_keys, var_options=options)
         varnames.extend(self._set_plugin_options('shell', variables, templar, task_keys))
 
         if self._connection.become is not None:
+            if self._play_context.become_pass:
+                # FIXME: eventually remove from task and play_context, here for backwards compat
+                # keep out of play objects to avoid accidental disclosure, only become plugin should have
+                # The become pass is already in the play_context if given on
+                # the CLI (-K). Make the plugin aware of it in this case.
+                task_keys['become_pass'] = self._play_context.become_pass
```

Comment:
this will not pickup ansible_<plugin>_pass variable overrides, what we need to do is pass the become(become_pass)/connection(password) pass to the 'set_options' of both plugins in the task_vars option (merging with existing task_vars dict)

Reply:
eventually we should just handle the passwords directly in task_worker vs sideloading via play_context, but that is a bigger change